src/Runner/Parallel/ParallelConfigFactory.php

```diff
@@ -47,11 +47,19 @@ public static function detect(
             ]);
         }
 
-        return new ParallelConfig(
-            ...array_filter(
-                [self::$cpuDetector->getCount(), $filesPerProcess, $processTimeout],
-                static fn ($value): bool => null !== $value
-            )
+        $args = array_filter(
+            [
+                'maxProcesses' => self::$cpuDetector->getCount(),
+                'filesPerProcess' => $filesPerProcess,
+                'processTimeout' => $processTimeout,
+            ],
+            static fn ($value): bool => null !== $value
         );
+
+        if (\PHP_VERSION_ID < 8_00_00) {
+            $args = array_values($args);
```

Comment:
This will "only" fix PHP 8+ arguments, as 7.4 will still move `processTimeout` argument into 2nd place, right?

Reply:
Yeah, I focused too much on named args, but even though I've tested it on PHP 7.4, I did not take `ParallelConfigFactory::detect(null, 360)` into consideration, where 360 would be shifted left and used as process per file, not as timeout. I've simplified it and factory now reuses `ParallelConfig`'s constants with default values.